pkg/controllermanager/controller/shoot/shoot_control.go

```diff
@@ -174,12 +175,24 @@ func (c *Controller) reconcileShootKey(key string) error {
 			c.recorder.Eventf(shoot, corev1.EventTypeNormal, "OperationPending", message)
 			c.updateShootStatusPending(shoot, message)
 		}
+	}
 
-	default:
-		// Otherwise (i.e., shoot is not ignored and may be reconciled) we start the reconcile operation).
-		needsRequeue, reconcileErr = c.control.ReconcileShoot(shoot, key)
+	defer c.scheduler.Done(shootElement.GetID())
+
+	// Create operation object
+	o, err := operation.New(shoot, shootLogger, c.k8sGardenClient, c.k8sGardenInformers.Garden().V1beta1(), c.identity, c.secrets, c.imageVector, c.config.ShootBackup)
+	if err != nil {
+		shootLogger.Errorf("Could not initialize a new operation: %s", err.Error())
+		return err
 	}
-	c.scheduler.Done(shootElement.GetID())
+
+	// Sync `Cluster` extension resource into the seed.
+	if err := o.SyncClusterResourceToSeed(context.TODO()); err != nil {
```

Comment:
Put the `context.TODO()` into a single `var` at the beginning of this method - this way, as soon as there is proper `context` support, you'll just need to change this single one instead of each call.